lib/cluster-providers/automode-cluster-provider.ts

```diff
@@ -0,0 +1,35 @@
+import { defaultOptions, GenericClusterProvider } from "./generic-cluster-provider";
+import * as eksv2 from "@aws-cdk/aws-eks-v2-alpha"
+import { IRole } from "aws-cdk-lib/aws-iam";
+
+export interface AutomodeClusterProviderProps extends Partial<eksv2.ClusterCommonOptions>{
+
+  id?: string,
+
+  /**
+   * @default: ['system,'general-purpose']
+   */
+  nodePools?: string[]
+
+  /**
+   * @default: generated by CDK
+   */
+  nodeRole?: IRole
+
+  tags?: {
+    [key: string]: string;
+  }
+
+}
+
+export class AutomodeClusterProvider extends GenericClusterProvider {
```

Comment:
It seems cleaner to create a GenericV2ClusterProvider first and then derive automode cluster provider from it. It creates a cleaner separation of concerns and less spagetti if/then checking is it v2 or v1 and/or automode. 
Happy to discuss.